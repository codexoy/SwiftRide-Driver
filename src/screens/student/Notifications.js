import React, { Fragment } from "react"
import {
	View,
	Text,
	TouchableHighlight,
	StyleSheet,
	FlatList,
	TouchableOpacity
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import PageTitle from "../../components/PageTitle"
import { MAIN_PADDING, NAME_LENGTH } from "../../consts"
import { Dropdown } from "react-native-material-dropdown"
import EmptyState from "../../components/EmptyState"
import StudentsLoader from "../../components/StudentsLoader"
import Notification from "../../components/Notification"
import moment from "moment"
import LessonPopup from "../../components/LessonPopup"
import Hours from "../../components/Hours"
import ShowReceipt from "../../components/ShowReceipt"

export class Notifications extends React.Component {
	static navigationOptions = () => {
		return {
			title: "notifications",
			tabBarLabel: strings("tabs.notifications_title"),
			tabBarAccessibilityLabel: strings("tabs.notifications_title"),
			tabBarTestID: "NotificationsTab"
		}
	}
	constructor(props) {
		super(props)
		this.filterOptions = [
			{
				value: "lessons/",
				label: strings("notifications.scheduled_lessons")
			},
			{
				value: "lessons/payments",
				label: strings("notifications.payments")
			}
		]
		this.state = {
			items: [],
			page: 1,
			nextUrl: "",
			filter: "",
			refreshing: false,
			visible: []
		}

		this._dropdownChange = this._dropdownChange.bind(this)
	}

	_onNavigationFocus = () => {
		this.setState(
			{
				filter:
					this.props.navigation.getParam("filter") ||
					this.filterOptions[0]["value"],
				loading: true,
				page: 1,
				nextUrl: ""
			},
			() => {
				this._getItems()
			}
		)
	}

	componentDidMount() {
		this.willFocusSubscription = this.props.navigation.addListener(
			"willFocus",
			payload => {
				this._onNavigationFocus()
			}
		)
	}

	componentWillUnmount() {
		this.willFocusSubscription.remove()
	}

	onRefresh = () => {
		this.setState({ page: 1, nextUrl: "", refreshing: true }, () => {
			this._getItems()
		})
	}

	_getItems = async (append = false) => {
		const resp = await this.props.fetchService.fetch(
			"/" +
				this.state.filter +
				"?limit=10&order_by=created_at desc&is_approved=true&creator_id=" +
				this.props.user.my_teacher.user.id +
				"&page=" +
				this.state.page,
			{
				method: "GET"
			}
		)
		let newValue = resp.json.data
		if (append) {
			newValue = [...this.state.items, ...newValue]
		}
		this.setState({
			items: newValue,
			nextUrl: resp.json["next_url"],
			loading: false,
			refreshing: false
		})
	}

	renderPayment = ({ item, index }) => {
		const date = moment
			.utc(item.created_at)
			.local()
			.format("DD.MM.YY, HH:mm")
		return (
			<Notification
				style={styles.notification}
				key={`payment${item.id}`}
				leftSide={<Text style={styles.amount}>{item.amount}â‚ª</Text>}
				basic={
					<Fragment>
						<Text style={styles.basic}>{date}</Text>
						<ShowReceipt
							item={item}
							dispatch={this.props.dispatch}
							user={this.props.user}
							style={{ alignSelf: "flex-start" }}
						/>
					</Fragment>
				}
			/>
		)
	}

	lessonPress = item => {
		let newVisible
		if (this.state.visible.includes(item.id)) {
			// we pop it
			newVisible = this.state.visible.filter((v, i) => v != item.id)
		} else {
			newVisible = [...this.state.visible, item.id]
		}
		this.setState({ visible: newVisible })
	}

	renderLesson = ({ item, index }) => {
		const visible = this.state.visible.includes(item.id) ? true : false
		return (
			<Fragment>
				<TouchableOpacity onPress={() => this.lessonPress(item)}>
					<Notification
						style={styles.notification}
						key={`lesson${item.id}`}
						basic={
							<Text style={styles.basic}>
								{this.props.user.my_teacher.user.name.slice(
									0,
									NAME_LENGTH
								)}{" "}
								{strings("notifications.teacher_scheduled")}
							</Text>
						}
						leftSide={
							<View>
								<Text style={styles.lessonDate}>
									{moment
										.utc(item.date)
										.local()
										.format("DD.MM")}
								</Text>
								<Text style={styles.lessonHour}>
									<Hours
										duration={item.duration}
										date={item.date}
									/>
								</Text>
							</View>
						}
					/>
					<LessonPopup
						visible={visible}
						item={item}
						onPress={this.lessonPress}
						testID="lessonPopup"
						navigation={this.props.navigation}
					/>
				</TouchableOpacity>
			</Fragment>
		)
	}

	_renderItems = () => {
		if (this.state.loading) {
			return <StudentsLoader />
		}
		let render = this.renderLesson
		let type = "lesson"
		if (this.state.filter.includes("payments")) {
			render = this.renderPayment
			type = "payment"
		}
		return (
			<FlatList
				data={this.state.items}
				renderItem={render}
				onEndReached={this.endReached.bind(this)}
				keyExtractor={item => `${type}${item.id}`}
				ListEmptyComponent={this._renderEmpty}
				extraData={this.state.visible}
				onRefresh={() => this.onRefresh()}
				refreshing={this.state.refreshing}
			/>
		)
	}

	endReached = () => {
		if (!this.state.nextUrl) return
		this.setState(
			{
				page: this.state.page + 1,
				nextUrl: ""
			},
			() => {
				this._getItems(true)
			}
		)
	}

	_dropdownChange = (value, index, data) => {
		this.setState(
			{
				page: 1,
				nextUrl: "",
				filter: value
			},
			() => {
				this._getItems()
			}
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="notifications"
			text={strings("empty_notifications")}
			style={styles.empty}
		/>
	)

	render() {
		return (
			<View style={styles.container}>
				<View testID="NotificationsView" style={styles.notifications}>
					<PageTitle
						style={styles.title}
						title={strings("tabs.notifications_title")}
						leftSide={
							<View>
								<Dropdown
									containerStyle={styles.dropdown}
									value={this.state.filter}
									data={this.filterOptions}
									onChangeText={this._dropdownChange}
									dropdownMargins={{ min: 0, max: 20 }}
									dropdownOffset={{ top: 0, left: 0 }}
									pickerStyle={{ marginTop: 60 }}
								/>
							</View>
						}
					/>
					<View style={styles.items}>{this._renderItems()}</View>
				</View>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	notifications: {
		flex: 1,
		paddingRight: MAIN_PADDING,
		paddingLeft: MAIN_PADDING,
		marginTop: 20
	},
	items: {
		marginTop: 20
	},
	notification: {},
	title: { marginBottom: 0 },
	dropdown: {
		alignSelf: "flex-end",
		width: 160,
		marginTop: 4
	},
	empty: {
		marginTop: 100
	},
	amount: {
		marginTop: 12,
		fontSize: 20,
		fontWeight: "bold"
	},
	basic: {
		marginTop: 10,
		fontSize: 18
	},
	lessonDate: {
		fontSize: 22,
		fontFamily: "Assistant-Light"
	},
	lessonHour: {
		fontWeight: "bold",
		fontSize: 14,
		color: "rgb(12, 116, 244)"
	}
})

function mapStateToProps(state) {
	return {
		user: state.user,
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(Notifications)
