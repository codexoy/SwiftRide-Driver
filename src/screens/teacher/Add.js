import React from "react"
import { View, Text, Button, StyleSheet, ScrollView } from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import Notification from "../../components/Notification"
import PageTitle from "../../components/PageTitle"
import NotificationButtons from "./NotificationButtons"

class Add extends React.Component {
	static navigationOptions = () => {
		return {
			title: "add",
			tabBarLabel: strings("tabs.add"),
			tabBarAccessibilityLabel: strings("tabs.add"),
			tabBarTestID: "AddTab"
		}
	}
	render() {
		return <ScrollView style={styles.container} />
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	notifications: {
		flex: 1,
		paddingRight: 30,
		paddingLeft: 30,
		marginTop: 20
	},
	notification: {}
})

export default connect()(Add)
