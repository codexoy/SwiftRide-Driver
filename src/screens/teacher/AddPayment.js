import { createStackNavigator } from "react-navigation"
import AddPaymentChooseStudent from "./AddPaymentChooseStudent"
import AddPaymentChooseAmount from "./AddPaymentChooseAmount"
export default createStackNavigator(
	{ First: AddPaymentChooseStudent, Second: AddPaymentChooseAmount },
	{
		initialRouteKey: "First",
		headerMode: "none",
		navigationOptions: {
			headerVisible: false
		}
	}
)
