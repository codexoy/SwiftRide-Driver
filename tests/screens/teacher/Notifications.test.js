import "react-native"
import React from "react"

import { Notifications } from "../../../src/screens/teacher/Notifications"

describe("Notifications", () => {
	test.skip("view renders correctly", done => {
		const student = { name: "test", student_id: 1 }
		fetch.mockResponseSuccess(
			JSON.stringify({
				data: [
					{
						duration: 40,
						student
					},
					{
						duration: 100,
						student
					}
				]
			})
		)
		const wrapper = shallow(
			<Notifications
				navigation={{
					...navigation,
					getParam: param => {
						return null
					},
					state: { params: { filter: "lessons" } }
				}}
				dispatch={dispatch}
				fetchService={fetchService}
			/>
		)
		testAsyncComponent(wrapper, done)
	})
})
