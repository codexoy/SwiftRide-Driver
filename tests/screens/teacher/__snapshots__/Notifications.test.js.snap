// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 20,
        "paddingLeft": 26,
        "paddingRight": 26,
      }
    }
    testID="NotificationsView"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 0,
          }
        }
      >
        התראות
      </Text>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "marginLeft": "auto",
            "marginTop": -5,
          }
        }
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "marginTop": 4,
                "width": 160,
              }
            }
          >
            <View
              accessible={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 4,
                  "right": 4,
                  "top": 6,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="box-only"
            >
              <View
                onResponderRelease={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="none"
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderBottomColor": "rgba(0, 0, 0, 0.38)",
                      "borderBottomWidth": 0.5,
                      "height": 31,
                      "paddingBottom": 8,
                      "paddingTop": -1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "position": "absolute",
                        "top": -17,
                      }
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.38)",
                          "fontSize": 12,
                        }
                      }
                    >
                      
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <TextInput
                      absoluteRTLLayout={false}
                      allowFontScaling={true}
                      autoCapitalize="sentences"
                      data={
                        Array [
                          Object {
                            "label": "בקשות שיעורים",
                            "value": "lessons/",
                          },
                          Object {
                            "label": "בקשות הצטרפות",
                            "value": "teacher/students",
                          },
                          Object {
                            "label": "פעילויות כספיות",
                            "value": "lessons/payments",
                          },
                        ]
                      }
                      disableFullscreenUI={true}
                      dropdownMargins={
                        Object {
                          "max": 20,
                          "min": 0,
                        }
                      }
                      dropdownOffset={
                        Object {
                          "left": 0,
                          "top": 0,
                        }
                      }
                      editable={false}
                      itemColor="rgba(0, 0, 0, .54)"
                      itemCount={4}
                      itemPadding={8}
                      labelExtractor={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onChangeText={[Function]}
                      onContentSizeChange={[Function]}
                      onFocus={[Function]}
                      propsExtractor={[Function]}
                      renderAccessory={[Function]}
                      rippleDuration={400}
                      selectionColor="rgb(0, 145, 234)"
                      shadeOpacity={0.12}
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "margin": 0,
                            "padding": 0,
                            "top": 2,
                          },
                          Object {
                            "color": "rgba(0, 0, 0, .87)",
                            "fontSize": 16,
                            "height": 24,
                            "textAlign": "left",
                          },
                          undefined,
                        ]
                      }
                      underlineColorAndroid="transparent"
                      useNativeDriver={false}
                      value="lessons"
                      valueExtractor={[Function]}
                    />
                    <View
                      style={
                        Object {
                          "alignSelf": "flex-start",
                          "justifyContent": "center",
                          "top": 2,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 24,
                            "justifyContent": "center",
                            "width": 24,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "height": 6,
                              "overflow": "hidden",
                              "width": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "height": 8,
                                  "transform": Array [
                                    Object {
                                      "translateY": -4,
                                    },
                                    Object {
                                      "rotate": "45deg",
                                    },
                                  ],
                                  "width": 8,
                                },
                                Object {
                                  "backgroundColor": "rgba(0, 0, 0, .38)",
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "height": 8,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "bottom": 0,
                          "left": 0,
                          "paddingVertical": 4,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <Text
                        numberOfLines={1}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "color": "rgb(213, 0, 0)",
                            "fontSize": 0,
                            "opacity": 0,
                          }
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "bottom": 0,
                          "left": 0,
                          "paddingVertical": 4,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <Text
                        numberOfLines={1}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "color": "rgba(0, 0, 0, .38)",
                            "fontSize": 12,
                            "opacity": 1,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View
                accessible={true}
                onLayout={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onRippleAnimation={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                rippleCentered={false}
                rippleColor="rgba(0, 0, 0, .38)"
                rippleDuration={400}
                rippleFades={true}
                rippleOpacity={0.54}
                rippleSequential={true}
                rippleSize={0}
                style={
                  Object {
                    "height": 48,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 16,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
            </View>
            <Modal
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              supportedOrientations={
                Array [
                  "portrait",
                  "portrait-upside-down",
                  "landscape",
                  "landscape-left",
                  "landscape-right",
                ]
              }
              transparent={true}
              visible={false}
            >
              <View
                onResponderRelease={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "rgba(255, 255, 255, 1.0)",
                        "borderRadius": 2,
                        "position": "absolute",
                        "shadowColor": "rgba(0, 0, 0, 1.0)",
                        "shadowOffset": Object {
                          "height": 2,
                          "width": 0,
                        },
                        "shadowOpacity": 0.54,
                        "shadowRadius": 2,
                      },
                      Object {
                        "height": 136,
                        "left": undefined,
                        "top": undefined,
                        "transform": Array [
                          Object {
                            "translateY": -48,
                          },
                        ],
                        "width": undefined,
                      },
                      Object {
                        "marginTop": 60,
                      },
                    ]
                  }
                >
                  <RCTScrollView
                    contentContainerStyle={
                      Object {
                        "paddingVertical": 8,
                      }
                    }
                    data={
                      Array [
                        Object {
                          "label": "בקשות שיעורים",
                          "value": "lessons/",
                        },
                        Object {
                          "label": "בקשות הצטרפות",
                          "value": "teacher/students",
                        },
                        Object {
                          "label": "פעילויות כספיות",
                          "value": "lessons/payments",
                        },
                      ]
                    }
                    disableVirtualization={false}
                    getItem={[Function]}
                    getItemCount={[Function]}
                    horizontal={false}
                    initialNumToRender={10}
                    keyExtractor={[Function]}
                    maxToRenderPerBatch={10}
                    numColumns={1}
                    onContentSizeChange={[Function]}
                    onEndReachedThreshold={2}
                    onLayout={[Function]}
                    onMomentumScrollEnd={[Function]}
                    onScroll={[Function]}
                    onScrollBeginDrag={[Function]}
                    onScrollEndDrag={[Function]}
                    renderItem={[Function]}
                    scrollEnabled={false}
                    scrollEventThrottle={50}
                    stickyHeaderIndices={Array []}
                    style={
                      Object {
                        "borderRadius": 2,
                        "flex": 1,
                      }
                    }
                    updateCellsBatchingPeriod={50}
                    viewabilityConfigCallbackPairs={Array []}
                    windowSize={21}
                  >
                    <View>
                      <View
                        onLayout={[Function]}
                        style={null}
                      >
                        <View
                          accessible={true}
                          focusAnimation={null}
                          focusAnimationDuration={225}
                          hitSlop={
                            Object {
                              "bottom": 6,
                              "left": 4,
                              "right": 4,
                              "top": 6,
                            }
                          }
                          onLayout={[Function]}
                          onPress={[Function]}
                          onPressIn={[Function]}
                          onPressOut={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          pointerEvents="box-only"
                          rippleCentered={false}
                          rippleColor="rgba(0, 0, 0, .38)"
                          rippleDuration={400}
                          rippleFades={true}
                          rippleOpacity={0.54}
                          rippleSequential={true}
                          rippleSize={0}
                          style={
                            Object {
                              "backgroundColor": "transparent",
                              "borderRadius": 0,
                              "height": 40,
                              "justifyContent": "center",
                              "paddingLeft": undefined,
                              "paddingRight": undefined,
                              "paddingVertical": 8,
                            }
                          }
                        >
                          <Text
                            numberOfLines={1}
                            style={
                              Array [
                                Object {
                                  "textAlign": "left",
                                },
                                undefined,
                                Object {
                                  "color": "rgba(0, 0, 0, .87)",
                                  "fontSize": 16,
                                },
                              ]
                            }
                          >
                            בקשות שיעורים
                          </Text>
                          <View
                            style={
                              Object {
                                "backgroundColor": "rgba(0, 0, 0, .38)",
                                "borderRadius": 0,
                                "bottom": 0,
                                "left": 0,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          />
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "bottom": 0,
                                  "left": 0,
                                  "overflow": "hidden",
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                },
                                Object {
                                  "borderRadius": 0,
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        onLayout={[Function]}
                        style={null}
                      >
                        <View
                          accessible={true}
                          focusAnimation={null}
                          focusAnimationDuration={225}
                          hitSlop={
                            Object {
                              "bottom": 6,
                              "left": 4,
                              "right": 4,
                              "top": 6,
                            }
                          }
                          onLayout={[Function]}
                          onPress={[Function]}
                          onPressIn={[Function]}
                          onPressOut={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          pointerEvents="box-only"
                          rippleCentered={false}
                          rippleColor="rgba(0, 0, 0, .38)"
                          rippleDuration={400}
                          rippleFades={true}
                          rippleOpacity={0.54}
                          rippleSequential={true}
                          rippleSize={0}
                          style={
                            Object {
                              "backgroundColor": "transparent",
                              "borderRadius": 0,
                              "height": 40,
                              "justifyContent": "center",
                              "paddingLeft": undefined,
                              "paddingRight": undefined,
                              "paddingVertical": 8,
                            }
                          }
                        >
                          <Text
                            numberOfLines={1}
                            style={
                              Array [
                                Object {
                                  "textAlign": "left",
                                },
                                undefined,
                                Object {
                                  "color": "rgba(0, 0, 0, .87)",
                                  "fontSize": 16,
                                },
                              ]
                            }
                          >
                            בקשות הצטרפות
                          </Text>
                          <View
                            style={
                              Object {
                                "backgroundColor": "rgba(0, 0, 0, .38)",
                                "borderRadius": 0,
                                "bottom": 0,
                                "left": 0,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          />
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "bottom": 0,
                                  "left": 0,
                                  "overflow": "hidden",
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                },
                                Object {
                                  "borderRadius": 0,
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        onLayout={[Function]}
                        style={null}
                      >
                        <View
                          accessible={true}
                          focusAnimation={null}
                          focusAnimationDuration={225}
                          hitSlop={
                            Object {
                              "bottom": 6,
                              "left": 4,
                              "right": 4,
                              "top": 6,
                            }
                          }
                          onLayout={[Function]}
                          onPress={[Function]}
                          onPressIn={[Function]}
                          onPressOut={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          pointerEvents="box-only"
                          rippleCentered={false}
                          rippleColor="rgba(0, 0, 0, .38)"
                          rippleDuration={400}
                          rippleFades={true}
                          rippleOpacity={0.54}
                          rippleSequential={true}
                          rippleSize={0}
                          style={
                            Object {
                              "backgroundColor": "transparent",
                              "borderRadius": 0,
                              "height": 40,
                              "justifyContent": "center",
                              "paddingLeft": undefined,
                              "paddingRight": undefined,
                              "paddingVertical": 8,
                            }
                          }
                        >
                          <Text
                            numberOfLines={1}
                            style={
                              Array [
                                Object {
                                  "textAlign": "left",
                                },
                                undefined,
                                Object {
                                  "color": "rgba(0, 0, 0, .87)",
                                  "fontSize": 16,
                                },
                              ]
                            }
                          >
                            פעילויות כספיות
                          </Text>
                          <View
                            style={
                              Object {
                                "backgroundColor": "rgba(0, 0, 0, .38)",
                                "borderRadius": 0,
                                "bottom": 0,
                                "left": 0,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          />
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "bottom": 0,
                                  "left": 0,
                                  "overflow": "hidden",
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                },
                                Object {
                                  "borderRadius": 0,
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </RCTScrollView>
                </View>
              </View>
            </Modal>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <View>
        <RNSVGCircle
          cx={316}
          cy="52"
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          r="22.75"
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height="6.4"
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          rx="4"
          ry="4"
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="70"
          x={210}
          y="36"
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height="6.4"
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          rx="3"
          ry="3"
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="40"
          x={240}
          y="50"
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height="6.4"
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          rx="3"
          ry="3"
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="40"
          x={240}
          y="64"
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height="10"
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          rx="0"
          ry="0"
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="12"
          x="14.84"
          y="40"
        />
        <RNSVGCircle
          cx={316}
          cy="124"
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          r="22.75"
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height="6.4"
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          rx="4"
          ry="4"
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="70"
          x={210}
          y="106"
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height="6.4"
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          rx="3"
          ry="3"
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="40"
          x={240}
          y="120"
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height="6.4"
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          rx="3"
          ry="3"
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="40"
          x={240}
          y="134"
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4278190080,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height="10"
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          rx="0"
          ry="0"
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="12"
          x="14.84"
          y="110"
        />
      </View>
    </View>
  </View>
</View>
`;
